pkgs/net-lib/net/server.rkt

```diff
@@ -0,0 +1,106 @@
+#lang racket/base
+
+(require racket/contract
+         racket/tcp)
+
+(provide
+ (contract-out
+  [start-server (server-proc/c (-> void?))]
+  [run-server (server-proc/c void?)]))
+
+(define (server-proc/c res/c)
+  (->* ((or/c #f string?)
+        listen-port-number?
+        (-> input-port? output-port? any))
+       (#:reuse? any/c
+        #:max-allow-wait exact-nonnegative-integer?
+        #:max-concurrent (or/c +inf.0 exact-positive-integer?)
+        #:listen-proc (-> listen-port-number? exact-nonnegative-integer? any/c (or/c #f string?) evt?)
+        #:accept-proc (-> any/c (values input-port? output-port?))
+        #:close-proc (-> any/c void?)
+        #:timeout-evt-proc (-> thread? input-port? output-port? boolean? evt?))
+       res/c))
+
+(define (start-server host port handle
+                      #:reuse? [reuse? #t]
+                      #:max-allow-wait [max-allow-wait 4]
+                      #:max-concurrent [max-concurrent +inf.0]
+                      #:listen-proc [listen tcp-listen]
+                      #:accept-proc [accept tcp-accept]
+                      #:close-proc [close tcp-close]
+                      #:timeout-evt-proc [make-timeout-evt (λ (_thd _in _out _break-sent?) never-evt)])
```

Comment:
I think I'd rather the keywords be `#:listen`, `#:accept`, and `#:close`. I'd also prefer `#:make-timeout-evt` to `#:timeout-evt-proc`, but maybe there's something even better.